test-data/unit/check-isinstance.test

```diff
@@ -2065,6 +2065,18 @@ def f() -> None:
 [builtins fixtures/dict.pyi]
 [out]
 
+[case testNarrowTypeAfterInDictKeys]
+# flags: --strict-optional
+from typing import Dict, Optional
+
+def f() -> None:
+   d: Dict[str, str] = {}
+   key: Optional[str]
+   if key not in d.keys():
```

Comment:
Let's change this test a bit: we need to make sure that `if key in d.keys()` narrows the type correctly.

Placing `reveal_type` inside `if` is the easiest way to do that.

Reply:
Testing `not in` is appreciated as well!

Reply:
> Let's change this test a bit: we need to make sure that `if key in d.keys()` narrows the type correctly.
> 
> Placing `reveal_type` inside `if` is the easiest way to do that.
> Testing not in is appreciated as well!

Updated [b5050c0](https://github.com/python/mypy/pull/13372/commits/b5050c00a1cdd7a7724929004bca076bd8cded4a)